pkg/loader/compose/compose.go

```diff
@@ -273,12 +322,131 @@ func loadPorts(composePorts []string) ([]kobject.Ports, error) {
 	return ports, nil
 }
 
-// LoadFile loads compose file into KomposeObject
+// LoadFile loads a compose file into KomposeObject
 func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
-	komposeObject := kobject.KomposeObject{
-		ServiceConfigs: make(map[string]kobject.ServiceConfig),
-		LoadedFrom:     "compose",
+
+	// Load the json / yaml file in order to get the version value
+	var version string
+
+	for _, file := range files {
+		composeVersion, err := getVersionFromFile(file)
+		if err != nil {
+			return kobject.KomposeObject{}, errors.Wrap(err, "Unable to load yaml/json file for version parsing")
+		}
+
+		// Check that the previous file loaded matches.
+		if len(files) > 0 && version != "" && version != composeVersion {
+			return kobject.KomposeObject{}, errors.New("All files passed must be the same version")
+		}
+		version = composeVersion
+	}
+
+	log.Debugf("Docker Compose version: %s", version)
+
+	// Convert based on version
+	switch version {
+	// Use libcompose for 1 or 2
+	// If blank, it's assumed it's 1 or 2
+	case "", "1", "1.0", "2", "2.0":
+		komposeObject, err := parseV1V2ComposeWithLibcompose(files)
```

Comment:
Yeah, since V1 / V2 is only supported by Libcompose. Makes sense to add it to the function name.